tests/utils/test_log_redact.py

```
@@ -858,3 +858,202 @@ def test_filter_installed_by_cli_main_callback(self) -> None:
+
+        # Shadow loguru._recattrs so the deferred `from loguru._recattrs import`
+        # inside _patcher raises ImportError.
+        fake_mod = types.ModuleType("loguru._recattrs")
+        monkeypatch.setitem(sys.modules, "loguru._recattrs", fake_mod)  # no RecordException attr
+
+        instance = CredentialRedactingFilter()
+        _install_on_loguru(instance)
+        patcher = instance._loguru_patcher  # type: ignore[attr-defined]
+
+        try:
+            raise ValueError("api_key=sk-secret")
```

Ruff (0.15.13)

[warning] Abstract `raise` to an inner function

(TRY301)